--- mdadm-3.2.7/mdadm.h.ftw~	2013-03-04 07:25:36.000000000 +0100
+++ mdadm-3.2.7/mdadm.h	2013-07-31 22:44:06.787194654 +0200
@@ -1070,9 +1070,11 @@ struct stat64;
 # ifndef __UCLIBC_HAS_LFS__
 #  define lseek64 lseek
 # endif
+# ifndef  __UCLIBC_HAS_NFTW__
+#  undef HAVE_NFTW
+# endif
 # ifndef  __UCLIBC_HAS_FTW__
 #  undef HAVE_FTW
-#  undef HAVE_NFTW
 # endif
 #endif
 
@@ -1092,7 +1094,7 @@ struct stat64;
 # endif
 #endif
 
-#ifdef HAVE_FTW
+#if defined(HAVE_FTW) || defined(HAVE_NFTW)
 # include <ftw.h>
 #endif
 
