--- mdadm-3.3/Makefile.0004~	2013-12-12 12:44:07.536123465 +0100
+++ mdadm-3.3/Makefile	2013-12-12 12:45:38.804037076 +0100
@@ -171,8 +171,13 @@ everything-test: all mdadm.static swap_s
 # mdadm.uclibc and mdassemble.uclibc don't work on x86-64
 # mdadm.tcc doesn't work..
 
+ifdef WHOLE_PROGRAM
+mdadm : check_rundir $(SRCS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -fwhole-program -Wl,--no-warn-common -flto -o mdadm $(SRCS) $(LDLIBS)
+else
 mdadm : check_rundir $(OBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o mdadm $(OBJS) $(LDLIBS)
+endif
 
 mdadm.static : $(OBJS) $(STATICOBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -static -o mdadm.static $(OBJS) $(STATICOBJS)
@@ -194,8 +199,13 @@ mdmon.O2 : $(MON_SRCS) $(INCL) mdmon.h
 	$(CC) -o mdmon.O2 $(CFLAGS) $(LDFLAGS) $(MON_LDFLAGS) -DHAVE_STDINT_H -O2 -D_FORTIFY_SOURCE=2 $(MON_SRCS)
 
 # use '-z now' to guarantee no dynamic linker interactions with the monitor thread
+ifdef WHOLE_PROGRAM
+mdmon : check_rundir $(MON_SRCS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MON_LDFLAGS) -Wl,-z,now -Wl,--no-warn-common -fwhole-program -flto -o mdmon $(MON_SRCS) $(LDLIBS)
+else
 mdmon : check_rundir $(MON_OBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) $(MON_LDFLAGS) -Wl,-z,now -o mdmon $(MON_OBJS) $(LDLIBS)
+endif
 msg.o: msg.c msg.h
 
 test_stripe : restripe.c xmalloc.o mdadm.h
