diff -up mdadm-4.3/Makefile.1~ mdadm-4.3/Makefile
--- mdadm-4.3/Makefile.1~	2024-03-03 12:01:07.685672168 +0100
+++ mdadm-4.3/Makefile	2024-03-03 13:01:50.795275194 +0100
@@ -50,7 +50,7 @@ ifeq ($(origin CC),default)
 CC := $(CROSS_COMPILE)gcc
 endif
 CXFLAGS ?= -ggdb
-CWFLAGS ?= -Wall -Werror -Wstrict-prototypes -Wextra -Wno-unused-parameter -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIE -Warray-bounds
+CWFLAGS ?= -Wall -Wstrict-prototypes -Wextra -Wno-unused-parameter -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIE -Warray-bounds
 ifdef WARN_UNUSED
 CWFLAGS += -Wp -O3
 endif
