--- mdadm-3.3.2/mdadm.h~	2015-04-19 01:29:36.045510483 +0200
+++ mdadm-3.3.2/mdadm.h	2015-04-19 01:30:37.344791483 +0200
@@ -111,7 +111,9 @@ extern __off64_t lseek64 __P ((int __fd,
  * some time include <linux/byteorder/xxx_endian.h> isn't enough,
  * and there is no standard conversion function so... */
 /* And dietlibc doesn't think byteswap is ok, so.. */
-/*  #include <byteswap.h> */
+#if defined(__UCLIBC__)
+#include <byteswap.h>
+#else
 #define bswap_16(x) (((x) & 0x00ffU) << 8 | \
 		     ((x) & 0xff00U) >> 8)
 #define bswap_32(x) (((x) & 0x000000ffU) << 24 | \
@@ -126,6 +128,7 @@ extern __off64_t lseek64 __P ((int __fd,
 		     ((x) & 0x0000ff0000000000ULL) >> 24 | \
 		     ((x) & 0x00000000ff000000ULL) << 8 | \
 		     ((x) & 0x000000ff00000000ULL) >> 8)
+#endif
 
 #if !defined(__KLIBC__)
 #if BYTE_ORDER == LITTLE_ENDIAN
